libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -72,9 +109,32 @@
  */
 #define UINT16_LOW_BYTE( x )     ( ( uint8_t ) ( ( x ) & 0x00ffU ) )
 
+/**
+ * @brief Macro for decoding a 2-byte unsigned int from a sequence of bytes.
+ *
+ * @param[in] ptr A uint8_t* that points to the high byte.
+ */
+#define UINT16_DECODE( ptr )                                \
+    ( uint16_t ) ( ( ( ( uint16_t ) ( *( ptr ) ) ) << 8 ) | \
+                   ( ( uint16_t ) ( *( ( ptr ) + 1 ) ) ) )
+
+/**
+ * @brief A value that represents an invalid remaining length.
+ *
+ * This value is greater than what is allowed by the MQTT specification.
+ */
+#define MQTT_REMAINING_LENGTH_INVALID               ( ( size_t ) 268435456 )
+
+/**
+ * @brief The minimum remaining length for a QoS 0 PUBLISH.
+ *
+ * Includes two bytes for topic name length and one byte for topic name.
+ */
+#define MQTT_MIN_PUBLISH_REMAINING_LENGTH_QOS0      ( 3U )
+
 /*-----------------------------------------------------------*/
 
-static size_t remainingLengthEncodedSize( size_t length )
+static size_t _remainingLengthEncodedSize( size_t length )
```

Comment:
These names will cause MISRA violations going forward. I get the idea of making internal functions start with underscore but this keeps coming up. If Public functions are prefixed with MQTT_ and private ones are not I think it is clear enough. We should document this in the style guide btw.

Reply:
After some discussion, we will be updating the style guide. I reverted this change and removed the leading underscore from function names